Doc/c-api/lifecycle.rst

```diff
@@ -0,0 +1,175 @@
+.. highlight:: c
+
+.. _life-cycle:
+
+Object Life Cycle
+=================
+
+Life Events
+-----------
+
+The figure below illustrates the order of events that can occur throughout an
+object's life [1]_.  An arrow from *A* to *B* indicates that event *B* can
+occur after event *A* has occurred, with the arrow's label indicating the
+condition that must be true for *B* to occur after *A*.
+
+.. raw:: html
+   :file: lifecycle.dot.svg
```

Comment:
Here's how it looks in dark mode:

<details>
<summary>Details</summary>

![image](https://github.com/user-attachments/assets/1af5d60c-9484-40d7-aab6-f38861284147)

</details>

Is it possible to make the SVG transparent and then apply `:class: invert-in-dark-mode`?

Compare https://docs.python.org/3/library/pathlib.html which is `.. image::` rather than `.. raw:: html`:

https://github.com/python/cpython/blob/75f7cf91ec5afc6091a0fd442a1f0435c19300b2/Doc/library/pathlib.rst#L21-L29

We should also have an alt tag.

Reply:
I got dark mode to work.  I didn't use `.. image::` because that breaks the clickable parts of the image.  SVG doesn't have an alt attribute, but I think I did the accessible equivalent.  I'm not sure how to test it though—any hints would be appreciated.